conference.js

```diff
@@ -1218,118 +1249,196 @@ export default {
+                                // Wait a little bit more just to be sure that
+                                // we won't miss the extension installation
+                                setTimeout(
+                                    () => {
+                                    DSExternalInstallationInProgress = false;
+                                    }, 500);
+                                APP.UI.removeListener(
+                                    UIEvents.EXTERNAL_INSTALLATION_CANCELED,
+                                    listener);
+                            };
+                            APP.UI.addListener(
+                                UIEvents.EXTERNAL_INSTALLATION_CANCELED,
+                                listener);
+                            APP.UI.showExtensionExternalInstallationDialog(url);
+                            break;
+                        }
+                        case "extensionFound": {
+                            if (externalInstallation) //close the dialog
+                                $.prompt.close();
+                            break;
+                        }
+                        default: {
+                            //Unknown status
                         }
                     }
                 }
-            }).then(([stream]) => {
-                // Stores the "untoggle" handler which remembers whether was
-                // there any video before and whether was it muted.
-                this._untoggleScreenSharing
-                    = this._turnScreenSharingOff
-                          .bind(this, didHaveVideo, wasVideoMuted);
-                DSExternalInstallationInProgress = false;
-                // close external installation dialog on success.
-                if(externalInstallation)
-                    $.prompt.close();
-                stream.on(
-                    TrackEvents.LOCAL_TRACK_STOPPED,
-                    () => {
-                        // If the stream was stopped during screen sharing
-                        // session then we should switch back to video.
-                        if (this.isSharingScreen){
-                            this._untoggleScreenSharing
-                                && this._untoggleScreenSharing();
-                        }
+            }
+        }).then(([desktopStream])=> {
+            // Stores the "untoggle" handler which remembers whether was
+            // there any video before and whether was it muted.
+            this._untoggleScreenSharing
+                = this._turnScreenSharingOff
+                      .bind(this, didHaveVideo, wasVideoMuted);
+            desktopStream.on(
+                TrackEvents.LOCAL_TRACK_STOPPED,
+                () => {
+                    // If the stream was stopped during screen sharing
+                    // session then we should switch back to video.
+                    if (this.isSharingScreen){
+                        this._untoggleScreenSharing
```

Comment:
you can combine this with the above if, in a single statement:

`this.isSharingScreen && this._untoggleScreenSharing && this._untoggleScreenSharing();`